WalletWasabi/WabiSabi/Client/Banning/CoinPrison.cs

```diff
@@ -78,6 +82,22 @@ private void ToFile()
 		File.WriteAllText(FilePath, json);
 	}
 
+	private DateTimeOffset ReduceBanningTimeIfNeeded(DateTimeOffset bannedUntil)
+	{
+		var currentDate = DateTimeOffset.UtcNow;
+		if (bannedUntil > currentDate.AddDays(_absoluteMaximumLocalPrisonBanningTimeInDays))
+		{
+			Random random = new();
+			int randomHours = random.Next(24, 49);
+			int randomMinutes = random.Next(0, 60);
+			int randomSeconds = random.Next(0, 60);
+
+			return currentDate.AddHours(randomHours).AddMinutes(randomMinutes).AddSeconds(randomSeconds);
```

Comment:
Won't this be the same reference? as in L88? Meaning here you are adding xHours xMinutes and xSeconds to an already existing 31 days?

Reply:
Nope